nanobind: leaked 1 types! causes cocotb to delete results.xml file

Hello, I found out about APyTypes at Orconf and I though it would be great to use it with cocotb! Unfortunately I ran into this error when running a simple cocotb test bench that uses APyTypes: https://github.com/JamesTimothyMeech/APyTypes-Test/blob/main/test/test.py. Is there anything that I am obviously doing wrong here?

```
james@james-Laptop-13th-Gen-Intel-Core:~/Desktop/APyTypes-Test/test$ make
rm -f results.xml
make -f Makefile results.xml
make[1]: Entering directory '/home/james/Desktop/APyTypes-Test/test'
mkdir -p sim_build/rtl
/usr/bin/iverilog -o sim_build/rtl/sim.vvp -D COCOTB_SIM=1 -s tb -I/home/james/Desktop/APyTypes-Test/test/../src -f sim_build/rtl/cmds.f -g2012   /home/james/Desktop/APyTypes-Test/test/../src/project.v /home/james/Desktop/APyTypes-Test/test/tb.v
rm -f results.xml
MODULE=test TESTCASE= TOPLEVEL=tb TOPLEVEL_LANG=verilog \
         /usr/bin/vvp -M /home/james/.local/lib/python3.10/site-packages/cocotb/libs -m libcocotbvpi_icarus   sim_build/rtl/sim.vvp 
     -.--ns INFO     gpi                                ..mbed/gpi_embed.cpp:76   in set_program_name_in_venv        Did not detect Python virtual environment. Using system-wide Python interpreter
     -.--ns INFO     gpi                                ../gpi/GpiCommon.cpp:101  in gpi_print_registered_impl       VPI registered
     0.00ns INFO     cocotb                             Running on Icarus Verilog version 11.0 (stable)
     0.00ns INFO     cocotb                             Running tests with cocotb v1.8.0 from /home/james/.local/lib/python3.10/site-packages/cocotb
     0.00ns INFO     cocotb                             Seeding Python random module with 1727639047
     0.00ns INFO     cocotb.regression                  Found test test.test_project
     0.00ns INFO     cocotb.regression                  running test_project (1/1)
     0.00ns INFO     cocotb.tb                          Start
     0.00ns INFO     cocotb.tb                          Reset
VCD info: dumpfile tb.vcd opened for output.
 90000.00ns INFO     cocotb.tb                          Test project behavior
110000.00ns INFO     cocotb.regression                  test_project passed
110000.00ns INFO     cocotb.regression                  **************************************************************************************
                                                        ** TEST                          STATUS  SIM TIME (ns)  REAL TIME (s)  RATIO (ns/s) **
                                                        **************************************************************************************
                                                        ** test.test_project              PASS      110000.00           0.00   70784511.23  **
                                                        **************************************************************************************
                                                        ** TESTS=1 PASS=1 FAIL=0 SKIP=0             110000.00           0.08    1308077.84  **
                                                        **************************************************************************************
                                                        
nanobind: leaked 1 types!
 - leaked type "apytypes._apytypes.APyFloat"
nanobind: leaked 53 functions!
 - leaked function "__add__"
 - leaked function "cast_to_half"
 - leaked function ""
 - leaked function "__gt__"
 - leaked function ""
 - leaked function ""
 - leaked function ""
 - leaked function "__rmul__"
 - leaked function "is_identical"
 - leaked function ""
 - leaked function ""
 - ... skipped remainder
nanobind: this is likely caused by a reference counting issue in the binding code.
make[1]: *** [/home/james/.local/lib/python3.10/site-packages/cocotb/share/makefiles/simulators/Makefile.icarus:95: results.xml] Aborted (core dumped)
make[1]: *** Deleting file 'results.xml'
make[1]: Leaving directory '/home/james/Desktop/APyTypes-Test/test'
make: *** [/home/james/.local/lib/python3.10/site-packages/cocotb/share/makefiles/Makefile.inc:40: sim] Error 2
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- Fix nanobind interaction with cocotb.
+
 ### Changed
 
 ### Removed

--- a/subprojects/nanobind.wrap
+++ b/subprojects/nanobind.wrap
@@ -1,9 +1,13 @@
-[wrap-git]
-method = cmake
-url = https://github.com/wjakob/nanobind/
-revision = v2.1.0
-depth = 1
-clone-recursive = True
+[wrap-file]
+directory = nanobind-2.1.0
+source_url = https://github.com/wjakob/nanobind/archive/refs/tags/v2.1.0.tar.gz
+source_filename = nanobind-2.1.0.tar.gz
+source_hash = c37c53c60ada5fe1c956e24bd4b83af669a2309bf952bd251f36a7d2fa3bacf0
+patch_filename = nanobind_2.1.0-1_patch.zip
+patch_url = https://wrapdb.mesonbuild.com/v2/nanobind_2.1.0-1/get_patch
+patch_hash = 79f6511b10627e75067efb883765d7b8865633185cd22db733c8816239b8dd6c
+source_fallback_url = https://github.com/mesonbuild/wrapdb/releases/download/nanobind_2.1.0-1/nanobind-2.1.0.tar.gz
+wrapdb_version = 2.1.0-1
 
 [provide]
-nanobind = nanobind_static_dep
+nanobind = nanobind_dep

--- a/subprojects/robin-map.wrap
+++ b/subprojects/robin-map.wrap
@@ -1,0 +1,13 @@
+[wrap-file]
+directory = robin-map-1.3.0
+source_url = https://github.com/Tessil/robin-map/archive/refs/tags/v1.3.0.tar.gz
+source_filename = robin-map-1.3.0.tar.gz
+source_hash = a8424ad3b0affd4c57ed26f0f3d8a29604f0e1f2ef2089f497f614b1c94c7236
+patch_filename = robin-map_1.3.0-1_patch.zip
+patch_url = https://wrapdb.mesonbuild.com/v2/robin-map_1.3.0-1/get_patch
+patch_hash = 6d090f988541ffb053512607e0942cbd0dbc2a4fa0563e44ff6a37e810b8c739
+source_fallback_url = https://github.com/mesonbuild/wrapdb/releases/download/robin-map_1.3.0-1/robin-map-1.3.0.tar.gz
+wrapdb_version = 1.3.0-1
+
+[provide]
+robin-map = robin_map_dep

